src/mpid/ch4/src/ch4_recv.h

```diff
@@ -373,7 +373,7 @@ MPL_STATIC_INLINE_PREFIX int MPID_Recv_init(void *buf,
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPID_RECV_INIT);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPID_RECV_INIT);
 
-    rreq = MPIR_Request_create_from_pool(MPIR_REQUEST_KIND__PREQUEST_RECV, 0);
+    rreq = MPIR_Request_create(MPIR_REQUEST_KIND__PREQUEST_RECV);
```

Comment:
I'm curious why the difference here vs. in the workq commit [6fc64009541a47a04ed0463d3f4041b918fd0e4b] which adds a VCI lock around `MPIR_Request_create`. There seems to be a contradiction with this statement:

> Now that we don't automatically add critical section inside
    MPIR_Request_create

Reply:
Ahh... good question! I did that early on to get `workq` out-of-my-way before the later iterations. We should either drop this commit or change the `MPIR_Request_create` into `MPIR_Request_create_pool`. I think the latter is probably better since the workq is inside ch4 and it will need use vci explicitly at some point.

Reply:
But your question was about the `prequest`, that is because prequest never enters device, so it doesn't need care about `vci`.

Reply:
> Ahh... good question! I did that early on to get `workq` out-of-my-way before the later iterations. We should either drop this commit or change the `MPIR_Request_create` into `MPIR_Request_create_pool`. I think the latter is probably better since the workq is inside ch4 and it will need use vci explicitly at some point.

Actually I did use `MPIR_Request_create_from_pool` for the workq in https://github.com/pmodels/mpich/pull/4499/commits/f837c3cdcb3c36f8ac46da120ee68b333a039927, so it is good :)

Reply:
There was some reasoning to differentiate prequest and workq -- mainly one is mpi-layer concept with less flexibility, and the other can be fully tweaked. However, in principle they have the same issues. Let me drop that commit and use explicit critical section in stead. That way, at least we have consistency inside `ch4`.

Reply:
@raffenet I dropped this commit and made all these request creation inside ch4 consistent, i.e. with explicit critical section and `MPIR_Request_create_from_pool`.